project_task_code/models/project_task.py

```diff
@@ -28,3 +29,40 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        code = self._name_search_parse_code(name)
+        tasks = []
+        if code:
+            # We are searching for codes
+            tasks = self.search(
+                [('code', '=ilike', code + '%')] + args,
+                limit=limit).name_get()
+            if limit and len(tasks) == limit:
+                return tasks
+        return tasks + super(ProjectTask, self).name_search(
+            name=name,
+            args=args,
+            operator=operator,
+            limit=(limit - len(tasks)) or limit)
+
+    @api.model
+    def _name_search_parse_code(self, name):
+        """ Check if user input looks like code """
+        # Remove any whitespace from name
+        name = name.lstrip()
+        # Get the data record
+        sequence_id = self.env.ref('project_task_code.sequence_task')
+        prefix = sequence_id.prefix or ''
+        # if user is typing only digits, return right away
+        if name.isdigit():
+            return prefix + name
+        # Check if input could be code
+        if (
+            name.startswith(prefix) or
+            name.lstrip(prefix).isdigit(),
```

Comment:
still do the slicing, `lstrip` strips any permutation of the characters in `prefix`